testnet/src/tenure.rs

```diff
@@ -40,91 +44,49 @@ impl <'a> Tenure {
+        microblock_pubkey.set_compressed(true);
+        let microblock_pubkeyhash = Hash160::from_data(&microblock_pubkey.to_bytes());
 
-        let block_builder = match burnchain_tip.block_snapshot.total_burn {
-            0 => StacksBlockBuilder::first(
-                1, 
-                &parent_block.metadata.burn_header_hash, 
-                parent_block.metadata.burn_header_timestamp, 
-                &vrf_proof, 
-                &microblock_secret_key),
-            _ => StacksBlockBuilder::from_parent(
-                1, 
-                &parent_block.metadata, 
-                &ratio, 
-                &vrf_proof, 
-                &microblock_secret_key)
-        };
+        let parent_block_total_burn = burnchain_tip.block_snapshot.total_burn;
 
         Self {
-            block_builder: Some(block_builder),
             coinbase_tx,
             config,
             burnchain_tip,
             mem_pool,
             parent_block,
             vrf_seed: VRFSeed::from_proof(&vrf_proof),
+            vrf_proof,
             burn_fee_cap,
-        }
-    }
-
-    pub fn handle_txs(&mut self, clarity_tx: &mut ClarityTx<'a>, txs_in_mempool: Vec<StacksTransaction>, candidates: &mut Vec<Txid>) {
-        for tx in txs_in_mempool {
-            if candidates.contains(&tx.txid()) {
-                continue;
-            }
-
-            let res = self.block_builder.as_mut()
-                .expect("BUG: attempted to process tx in tenure that already committed")
-                .try_mine_tx(clarity_tx, &tx);
-            match res {
-                Err(e) => error!("Failed mining transaction - {}", e),
-                Ok(_) => {
-                    candidates.push(tx.txid());
-                },
-            };
+            microblock_pubkeyhash,
+            parent_block_total_burn
         }
     }
 
     pub fn run(&mut self) -> Option<TenureArtifacts> {
         info!("Node starting new tenure with VRF {:?}", self.vrf_seed);
 
-        let mut chain_state = StacksChainState::open_with_block_limit(
-            false, 
-            TESTNET_CHAIN_ID, 
-            &self.config.get_chainstate_path(),
-            self.config.block_limit.clone()).unwrap();
-
-        let burn_header_hash = self.parent_block.metadata.burn_header_hash;
-        let block_hash= self.parent_block.block.block_hash();
-
-        let mut clarity_tx = self.block_builder.as_mut()
-            .expect("BUG: attempted to process tx in tenure that already committed")
-            .epoch_begin(&mut chain_state).unwrap();
-        let mut candidates = vec![];
-
-        self.handle_txs(&mut clarity_tx, vec![self.coinbase_tx.clone()], &mut candidates);
-
         let duration_left: u128 = self.config.burnchain.commit_anchor_block_within as u128;
         let mut elapsed = Instant::now().duration_since(self.burnchain_tip.received_at);
         while duration_left.saturating_sub(elapsed.as_millis()) > 0 {
```

Comment:
Should we just get rid of this loop as well, and have one simple sleep?

Reply:
I'm happy enough leaving this in, but if you want, I can take it out.